lib/Dialect/FIRRTL/FIRRTLFolds.cpp

```diff
@@ -1264,11 +1264,93 @@ class MuxPad : public mlir::RewritePattern {
     return success();
   }
 };
+
+// Find muxes which have conditions dominated by other muxes with the same
+// condition.
+class MuxSharedCond : public mlir::RewritePattern {
+public:
+  MuxSharedCond(MLIRContext *context)
+      : RewritePattern(MuxPrimOp::getOperationName(), 0, context) {}
+
+  Value updateOrClone(MuxPrimOp mux, Value high, Value low,
+                      mlir::PatternRewriter &rewriter, bool chain) const {
```

Comment:
`chain` may be more obvious if named `safeToUpdateInPlace` or something more descriptive.